plugins/modules/katello_repository_set.py

```diff
@@ -152,7 +152,6 @@
 def get_desired_repos(desired_substitutions, available_repos):
     desired_repos = []
     for sub in desired_substitutions:
-        sub = {k: v for (k, v) in sub.items() if v is not None}
         desired_repos += filter(lambda available: available['substitutions'] == sub, available_repos)
```

Comment:
Does this work in Python 3? Isn't this a `list += generator`?

Reply:
It does work :)